### ISSUE TITLE
[L4D2] Steam group servers not showing up

### ISSUE THREAD

**Issue by lDrDooml** [ValveSoftware/Source-1-Games#6007]
With the new update `2.2.3.8`, the function that is responsible for displaying and loading the servers' visualization stopped working, the servers simply no longer respond to the requests coming from the client.

I hope this was simply some mistake and not intentional because if so the community servers are going to disappear for the simple fact that no one is going to want to rent an empty host. 

Solutions have been offered such as the server browser function that the game offers but it does not work correctly and is not even added to the game menu, so it is a terrible solution because the vast majority of players are ignorant in this matter and connect to the first server they see.

**Comment by argonatorUberTF** [comment]
Yes, I second the reporting of this issue.  It appeared between April to May 15th within those steam updates.
I haven't play for a while and now it is not there.
There used to be a hot list of known servers that you join for the Turbo Left Dead group. Now it's gone.
However the only workaround is to maintain a list separately on the sophisticated application called notepad.exe and use cut and paste into the console to connect to said IP.  I can't fathom why Steam can't do something of this high caliber that users can do manually with a 20 year old Microsoft product manually.
The server favorites in the connection manager doesn't retain the IP addresses also.

**Comment by bkdwt** [comment]
Same problem here.

**Comment by DJHuck** [comment]
This menu let's you access the Server Browser and set favorites: https://steamcommunity.com/sharedfiles/filedetails/?id=2389776402 right from the main menu of the game. Shows you EVERY server online in the universe. Plus tracks your history of playing, does the server browser. Been using it for 10 years. It will get you through until Valve fixes things.

You could also bind this in the console to get to your specific Steam group server: bind "F8" "mm_dedicated_force_servers 74.91.114.122:27015,74.91.126.26:27015" When in a lobby you press F8 after setting up game with Best Available, and the lobby will go to one of those two Steam Group servers. You can list as many servers as you need to, you don't have to use my group servers.

**Comment by lDrDooml** [comment]
> Replying to https://github.com/ValveSoftware/Source-1-Games/issues/6007#issuecomment-2119524132

The information is appreciated, but the point here is something else and that is that even if there are alternatives to connect to the community servers, the number of players who find out about this is not even **2%** or **3%** of the total of average players of the game, to get a better idea I have a tool on my website for my servers that allows me to see the average number of new players that enter per day and from **150** to **10**, which is catastrophic.

I have seen people who are happy about this situation but we and the vast majority are not to blame for the fact that this function has been abused by the Chinese servers or the Russian servers abusing the matchmaking system, forcing players who were playing on official servers to connect to their servers.

If there is a lack of solutions, for example, it could be implemented that the servers are displayed according to the number of ping, for example if it has more than **300** ms it will not be displayed for logical reasons. If not, the other thing is that **L4D2** can use the token system that **CSGO** had **(I don't know if CS2 has it)**.

**Comment by cairnsw** [comment]
I'm having the same problem, using bind and forcing a dedicated server, but I agree that is not ideal.  The group server list is far more convenient for my users.

**Comment by SCP-076** [comment]
The issue has not been fully fixed after today's update, and I have explained the reason here:
https://github.com/Tsuey/L4D2-Community-Update/issues/518

**Comment by ThingOfNull** [comment]
As of today (6/30/24), the latest server version 2.2.4.0 still does not appear to have fixed this issue.

**Comment by ThingOfNull** [comment]
By the way, the 'sv_password" parameter also seems to not work properly, as it is unable to set a password.

https://github.com/ValveSoftware/Source-1-Games/issues/3416
